screens/map/MapScreen.js

```diff
@@ -35,198 +35,100 @@ import {
 
 const snapPoints = [185, 325, 488];
 
-const deltas = {
-  latitudeDelta: 0.01,
-  longitudeDelta: 0.01,
-};
-
-export default class MapScreen extends React.Component {
-  constructor(props) {
-    super(props);
+export default function MapScreen(props) {
+  const [region, setRegion] = useState(initialRegion);
+  const [currentStore, setCurrentStore] = useState(null);
 
-    this.state = {
-      locationErrorMsg: null,
-      region: initialRegion,
-      stores: null,
-      store: null,
-      storeProducts: null,
-      showDefaultStore: false,
-    };
-  }
+  const bottomSheetRef = useRef(null);
+  const mapRef = useRef(null);
+  const storeProducts = useStoreProducts(currentStore);
+  const { locationPermissions, initialLocation } = useCurrentLocation();
+  let stores = useStores();
 
-  async componentDidMount() {
-    // We get current location first, since we need to use the lat/lon found in _populateIntitialStoresProducts
-    await this._findCurrentLocation();
-    await this._populateInitialStoresProducts();
+  if (locationPermissions === 'granted') {
+    stores = orderStoresByDistance(initialLocation, stores);
```

Comment:
Ok I think I'm still a little confused about how I should handle this. Like you mentioned, `orderStoresByDistance` is neither expensive nor async, and my initial `useSortedStores` mess was totally unnecessary. Still, I don't think I fully understood what you meant in [your comment from earlier](https://github.com/calblueprint/dccentralkitchen/pull/184#discussion_r549527181) about turning this into a transformation that can be done on-the-fly. Where/when/how would I do that? 😅 

Also, even if it isn't expensive/async, do you think it'd still be a good idea to use `useMemo` just to prevent this from running on every render? I realized this is modifying `stores` every time it runs (though I'm not sure why since the order/distances shouldn't be changing), which means it's also triggering the `useEffect` below which has `stores` as a dependency, even though that should only be executed once.

Reply:
You are doing it! On-the-fly meaning somewhere in the render path (or right before it) — just not within an effect. I see that you need the ordered stores within your effect below, though. Since `orderStoresByDistance` returns a new array every time (which is fine), I think memoizing as an optimization to prevent unexpected updates could be a good idea.

I would also consider not overshadowing the `stores` name with the ordered stores value to aid own debugging with the changing values 😄 

> though I'm not sure why since the order/distances shouldn't be changing

Changing values in React always use a "shallow equal" comparison (does it pass `===`?), so new arrays with identical contents will be considered different (i.e., `[] === [] -> false`)